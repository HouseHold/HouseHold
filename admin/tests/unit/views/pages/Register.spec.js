import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import CoreuiVue from '@coreui/vue'
import Register from '@/views/pages/Register'

Vue.use(CoreuiVue)

describe('Register.vue', () => {
  it('has a name', () => {
    expect(Register.name).toBe('Register')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Register)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Register', () => {
    const wrapper = shallowMount(Register)
    expect(wrapper.is(Register)).toBe(true)
  })
  it('should render correct content', () => {
    const wrapper = shallowMount(Register)
    expect(wrapper.find('h1').text()).toMatch('Register')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Register)
    expect(wrapper.element).toMatchSnapshot()
  })
})
