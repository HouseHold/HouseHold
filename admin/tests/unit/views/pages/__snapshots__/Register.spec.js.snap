// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register.vue renders correctly 1`] = `
<ccontainer-stub
  class="min-vh-100 d-flex align-items-center"
  tag="div"
>
  <crow-stub
    class="w-100 justify-content-center"
    gutters="true"
    tag="div"
  >
    <ccol-stub
      md="6"
      sm="8"
      tag="div"
    >
      <ccard-stub
        class="mx-4 mb-0"
      >
        <ccardbody-stub
          class="p-4"
        >
          <cform-stub>
            <h1>
              Register
            </h1>
             
            <p
              class="text-muted"
            >
              Create your account
            </p>
             
            <cinput-stub
              autocomplete="username"
              lazy="400"
              placeholder="Username"
              type="text"
            />
             
            <cinput-stub
              autocomplete="email"
              lazy="400"
              placeholder="Email"
              prepend="@"
              type="text"
            />
             
            <cinput-stub
              autocomplete="new-password"
              lazy="400"
              placeholder="Password"
              type="password"
            />
             
            <cinput-stub
              autocomplete="new-password"
              class="mb-4"
              lazy="400"
              placeholder="Repeat password"
              type="password"
            />
             
            <cbutton-stub
              activeclass="active"
              block="true"
              color="success"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              Create Account
            </cbutton-stub>
          </cform-stub>
        </ccardbody-stub>
         
        <ccardfooter-stub
          class="p-4"
        >
          <crow-stub
            gutters="true"
            tag="div"
          >
            <ccol-stub
              col="6"
              tag="div"
            >
              <cbutton-stub
                activeclass="active"
                block="true"
                color="facebook"
                event="click"
                exactactiveclass="active"
                routertag="a"
                target="_self"
                type="button"
              >
                
                Facebook
              
              </cbutton-stub>
            </ccol-stub>
             
            <ccol-stub
              col="6"
              tag="div"
            >
              <cbutton-stub
                activeclass="active"
                block="true"
                color="twitter"
                event="click"
                exactactiveclass="active"
                routertag="a"
                target="_self"
                type="button"
              >
                
                Twitter
              
              </cbutton-stub>
            </ccol-stub>
          </crow-stub>
        </ccardfooter-stub>
      </ccard-stub>
    </ccol-stub>
  </crow-stub>
</ccontainer-stub>
`;
